packages/Caching/Detector/ChangedFilesDetector.php

```diff
@@ -24,13 +34,34 @@ public function __construct(
     /**
      * @param string[] $dependentFiles
      */
-    public function addFileWithDependencies(string $filePath, array $dependentFiles): void
+    public function addFileDependentFiles(string $filePath, array $dependentFiles): void
+    {
+        $filePathCacheKey = $this->getFilePathCacheKey($filePath);
+        $this->dependentFiles[$filePathCacheKey] = $dependentFiles;
+    }
+
+    public function cacheFileWithDependencies(string $filePath): void
     {
         $filePathCacheKey = $this->getFilePathCacheKey($filePath);
+
+        if (! array_key_exists($filePathCacheKey, $this->cachableFiles)) {
```

Comment:
probably `isset()` can be used?

Reply:
I tend to favor being precise using `array_key_exists` when I want to check for the existance of a key.
`isset` will work for sure here as values of the array are anyway always `true`
so up to you